arbnode/node.go

```diff
@@ -823,23 +823,26 @@ func CreateNode(
 }
 
 func (n *Node) Start(ctx context.Context) error {
-	execClient, ok := n.Execution.(*gethexec.ExecutionNode)
-	if !ok {
-		execClient = nil
+	executionNode, isExecutionNode := n.Execution.(*gethexec.ExecutionNode)
+	executionClientImpl, isExecutionClientImpl := n.Execution.(*gethexec.ExecutionClientImpl)
```

Comment:
this is o.k as first approximation. However, I have a different destination in mind:
I want to have two separate New() functions, the existing one and one that accepts an ExecutionClient interface instead of a FullExecutionClient. Both should obviously share most code.
The version that accepts an ExecutionClient interface should  make sure that batch poster & sequencer are disabled. Both should create the same Node struct. Possibly some new nil checks are required (I think that the Node struct should have a nil for Execution but not sure).
This can be done here or in a separate PR.

Reply:
that way would not require an ExecutionClientImpl, since the interface would be limited to the supported functions.